### ISSUE TITLE
HexValidation incorrectly validates on strings

### ISSUE THREAD

**Issue by srawlins** [brigade/scss-lint#597]
```shell
$ cat hex.scss
polyfill-unscoped-rule {
  content: 'div#ab-overlay-container';
}
$ bundle exec bin/scss-lint hex.scss
hex.scss:2 [W] HexValidation: Colors must have either three or six digits: `#ab`
```

HexValidation should not lint on quoted strings, ever, right?

**Comment by sds** [comment]
I can't think of a situation where you _need_ a string to pass a hex color value, but it's perfectly valid to do so, and I could see some developers doing so so their syntax highlighter handles the color correctly.

The situation you describe above, however, is definitely something we should fix. In that case we can easily tell a color was not intended because there are no spaces on either side of the matched hex value.

**Comment by srawlins** [comment]
OK this is supremely annoying. We can't even really use `\b` or smoething similar, because you can do arithmetic on Hex colors, even without whitespace:

```
$ bundle exec sass --scss
p {
  color: #010203+#040506;
}
p {
  color: #050709; }
$ bundle exec sass --scss
p {
  color: #010203-#040506;
}
p {
  color: black; }
$ bundle exec sass --scss
p {
  color: #000000+red;
}
p {
  color: red; }
```

Hrm. So `#ab-overlay` could totally look like it was _supposed_ to be a color.

**Comment by srawlins** [comment]
That last comment was hasty: it turns out that Sass will blow up if you try to add two things that aren't proper hex, like `#1234+#5678`. So we don't have to worry about those cases.

**Comment by sds** [comment]
Fixed in #610.
